File: gallery/src/main/java/com/qkopy/gallery/ui/imagepicker/ImagePickerView.java
/*
 * Created by @ajithvgiri on 4/12/18 8:14 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 4/12/18 7:05 PM
 */

package com.qkopy.gallery.ui.imagepicker;

import com.qkopy.gallery.model.Folder;
import com.qkopy.gallery.model.Image;
import com.qkopy.gallery.ui.common.MvpView;

import java.util.List;

public interface ImagePickerView extends MvpView {

    void showLoading(boolean isLoading);

    void showFetchCompleted(List<Image> images, List<Folder> folders);

    void showError(Throwable throwable);

    void showEmpty();

    void showCapturedImage(List<Image> images);

    void finishPickImages(List<Image> images);

}

File: gallery/src/main/java/com/qkopy/gallery/ui/imagepicker/ImagePickerView.kt
package com.qkopy.gallery.ui.imagepicker

import com.qkopy.gallery.model.Folder
import com.qkopy.gallery.model.Image
import com.qkopy.gallery.ui.common.MvpView

interface ImagePickerView : MvpView {
    fun showLoading(isLoading: Boolean)
    fun showFetchCompleted(
        images: List<Image>,
        folders: List<Folder>
    )

    fun showError(throwable: Throwable?)
    fun showEmpty()
    fun showCapturedImage(images: List<Image>?)
    fun finishPickImages(images: List<Image>?)
    fun showFetching(images: List<Image>?,folders: List<Folder>?)
    fun showUpdateFolder(folder: Folder)
    fun showUpdateImage(images: List<Image>?)
}